### ISSUE TITLE
开新坑，x-ui 面板

### ISSUE THREAD

**Issue by sprov065** [sprov065/v2-ui#469]
目前 python 版的 v2-ui 维护起来较为困难，扩展性也不足，python 编译出来的二进制文件兼容性差，时不时可能在某些系统上无法运行，所以打算使用 go 语言开发一个新的面板，名字暂时叫做 x-ui，以下是项目地址：

https://github.com/sprov065/x-ui

go 语言的好处自不用多说了，性能好，内存占用低，可以直接编译各个平台的二进制文件等等

x-ui 目前打算先将 v2-ui 的绝大部分功能实现，并提供简易的从 v2-ui 迁移到 x-ui 到方式，之后再继续开发 x-ui 的新功能，v2-ui 的使命将会在 x-ui 稳定后画上句号。

目前 x-ui 处于开发初期，如果你有好的想法与设计，欢迎贡献你的建议以及 pr

**Comment by BlueSkyXN** [comment]
惨python惨

**Comment by BlueSkyXN** [comment]
建议加一个自定义路由规则功能上去，发挥Ui优势
（比如直接block某法）

**Comment by sprov065** [comment]
> 建议加一个自定义路由规则功能上去，发挥Ui优势
> （比如直接block某法）

可以有，路由的众多灵活配置对于 ui 来说实现起来比较复杂，可以挑典型的功能加进 ui

**Comment by anmonick** [comment]
大佬，可不可以加个多服务器负载均衡啊（速度叠加）

**Comment by sprov065** [comment]
提建议的时候也麻烦搞清楚 xray 是什么，xray 有什么功能，面板能做什么事，不要随便想到什么就说

**Comment by anmonick** [comment]
> 大佬，可不可以加个多服务器负载均衡啊（速度叠加）

![20210407222208-530x200](https://user-images.githubusercontent.com/81863021/119989555-e5e9d180-bff9-11eb-8879-834ef8bd8e45.jpg)

**Comment by 2417559113** [comment]
希望功能上支持单端口多用户

**Comment by OniquaJliu** [comment]
xray可以通过回落机制实现多协议的端口复用，新面板会很好的实现这个功能吗？

**Comment by ysslang** [comment]
希望支持docker部署

**Comment by lynkco01** [comment]
> 希望可以新增MTProto...

MTProto 现在很容易被封，只有在加了白名单的情况下使用

**Comment by wdsfto** [comment]
6666666666666




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年5月28日(星期五) 下午2:07
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; 
主题: Re: [sprov065/v2-ui] 开新坑，x-ui 面板 (#469)





  
建议加一个自定义路由规则功能上去，发挥Ui优势
 （比如直接block某法）
  
可以有，路由的众多灵活配置对于 ui 来说实现起来比较复杂，可以挑典型的功能加进 ui

**Comment by OniquaJliu** [comment]
希望可以添加多ssl证书

**Comment by wuhansusu** [comment]
支持大大

**Comment by kobehuanghai** [comment]
希望新坑可以支持单端口多用户，目前用vless+xtls多用户的遗憾

**Comment by yoyoll** [comment]
好的 谢谢大佬

**Comment by yoyoll** [comment]
> > 建议加一个自定义规则功能介绍，发挥你的优势
> > （比如直接阻止某法）
> 
> 可以有，语音的动态配置对于ui来说简单的实现起来比较复杂，可以添加典型的功能加ui



> > 建议加一个自定义路由规则功能上去，发挥Ui优势
> > （比如直接block某法）
> 
> 可以有，路由的众多灵活配置对于 ui 来说实现起来比较复杂，可以挑典型的功能加进 ui

作者大大 我对这持不赞同的看法 当然这功能的实现可能非常有用 但对于不属于block内容或block掉某个域名而造成其他web需要加载到这个的就出现缺失加载不出的情况  让不明白的以为作者的脚本有不通的地方  当然 实现这个功能出发点是好的  我想可能会带给作者困扰和对作者的不理解  个人用户一般就那么几个节点  真上高级就直接soga xui的在于简便易于上手  以上仅是个人想法  如有不对  各位大佬 板砖小点 哈哈

**Comment by sprov065** [comment]
> > > 建议加一个自定义规则功能介绍，发挥你的优势
> > > （比如直接阻止某法）
> > 
> > 
> > 可以有，语音的动态配置对于ui来说简单的实现起来比较复杂，可以添加典型的功能加ui
> 
> > > 建议加一个自定义路由规则功能上去，发挥Ui优势
> > > （比如直接block某法）
> > 
> > 
> > 可以有，路由的众多灵活配置对于 ui 来说实现起来比较复杂，可以挑典型的功能加进 ui
> 
> 作者大大 我对这持不赞同的看法 当然这功能的实现可能非常有用 但对于不属于block内容或block掉某个域名而造成其他web需要加载到这个的就出现缺失加载不出的情况 让不明白的以为作者的脚本有不通的地方 当然 实现这个功能出发点是好的 我想可能会带给作者困扰和对作者的不理解 个人用户一般就那么几个节点 真上高级就直接soga xui的在于简便易于上手 以上仅是个人想法 如有不对 各位大佬 板砖小点 哈哈

我的意思只是提供设置路由规则的功能，并不是提供默认的一些路由规则

**Comment by yoyoll** [comment]
是的 我明白您的意思  怕有人自己block掉域名 导致用着过程 出现加载不出  有这么例子 block掉某个域名的网址 tiktok的头像和视频加载不出   想说自己block掉自己 可能还不知 网络的情况使用千千万万 不知道别人会好奇block掉啥  然后又不通  就说脚本问题

**Comment by hide-on-bush-lalala** [comment]
我发现一个问题，就是面板创建的账户好像默认优先使用ipv6，这导致我原生解锁Netflix的机器无法观看Netflix。
Netflix现在ban了很多很多的ipv6，我也没有看到面板在哪里设置优先使用ipv4，希望作者可以增加这个按钮

**Comment by sprov065** [comment]
> 我发现一个问题，就是面板创建的账户好像默认优先使用ipv6，这导致我原生解锁Netflix的机器无法观看Netflix。
> Netflix现在ban了很多很多的ipv6，我也没有看到面板在哪里设置优先使用ipv4，希望作者可以增加这个按钮

面板没有配置的都属于 xray-core 的默认行为，类似这种特殊需求的，自行修改 xray 配置模版中的 dns 配置

**Comment by hide-on-bush-lalala** [comment]
> > 我发现一个问题，就是面板创建的账户好像默认优先使用ipv6，这导致我原生解锁Netflix的机器无法观看Netflix。
> > Netflix现在ban了很多很多的ipv6，我也没有看到面板在哪里设置优先使用ipv4，希望作者可以增加这个按钮
> 
> 面板没有配置的都属于 xray-core 的默认行为，类似这种特殊需求的，自行修改 xray 配置模版中的 dns 配置

是xray-core的默认行为，但是还不知道怎么配置，想问下作者xray-core的配置文件路径在哪里....

面板那个配置模板如果改错了（由于我对xray-core的配置还很不熟悉），面板直接crash，而且启动不起来，我只会重装解决...

**Comment by yoyoll** [comment]
大佬 既然有流量限制 后续会不会出到期日自动禁用功能

**Comment by yoyoll** [comment]
大佬 是不是加个总流量重置和流量到期日重置功能会更好

**Comment by Zzz3m** [comment]
有些用户是x-ui配合将tls交给nginx实现的。那么在面板生成二维码和链接的时候，监听端口，地址和tls，与实际的并不相符。希望可以能自定义（手动填写）端口，地址和tls，这三个信息来生成对应的准确的二维码和链接。

**Comment by 2417559113** [comment]
后期支持单端口多用户么？因为这样可以解决中专机的端口麻烦。

**Comment by Zzz3m** [comment]
> 后期支持单端口多用户么？因为这样可以解决中专机的端口麻烦。

配合nginx可以实现单端口多用户。但是需要自己配置。

**Comment by 2417559113** [comment]
> > 后期支持单端口多用户么？因为这样可以解决中专机的端口麻烦。
> 
> 配合nginx可以实现单端口多用户。但是需要自己配置。

配合nginx可以实现单端口多用户，需要自己配置，可以出一个教程么？因为有很多人像我这种小白都不会方法。

**Comment by XiaoFeiOS** [comment]
有限速功能吗

**Comment by evin792** [comment]
大佬能否添加定时任务功能？有时候经常需要重启才能继续使用，如果可以设置自定义时间重启xray，那就美滋滋
